messaging/WFLY-13797_galleon_layer_for_remote_artemis.adoc

```diff
@@ -0,0 +1,123 @@
+:author:            Brian Stansberry
+:email:             brian.stansberry@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The Jakarta Messaging Service (JMS) provides an API for sending messages between two or more clients and a programming model to handle the producer-consumer messaging problem.
+Users wishing to use JMS in their applications need a mechanism to connect to a message broker which is responsible for accepting messages from producers and making them available
+to consumers. The WildFly server is responsible for providing that mechanism to applications in accordance with the JMS specification. There are a number of ways the server
+can be configured to provide the necessary facilities. A good practice is use an external-to-the-appserver broker (aka a remote broker), allowing a separation of administrative concerns between
+managing the broker and managing the application. An ActiveMQ Artemis server is a good choice for a remote broker, particularly for WildFly users who may be familiar with
+ActiveMQ Artemis due to WildFly's own use of it as it's embedded broker. So, WildFly should make it easy for users to provision a server that can integrate with a remote ActiveMQ broker.
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] which will be responsible for providing the necessary remote ActiveMQ broker integration.
+
+The following is a configuration example of the WildFly bootable JAR Maven plugin to provision a server with the `remote-activemq` layer:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <configuration>
+        <feature-pack-location>wildfly@maven(org.jboss.universe:community-universe)#${version.wildfly}</feature-pack-location>
+        <layers>
+            <layer>remote-activemq</layer>
+        </layers>
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>package</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
+----
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13797[WFLY-13797 Add a Galleon layer for messaging integration with a remote ActiveMQ broker]
+* https://issues.redhat.com/browse/EAP7-1549[EAP7-1549 Add a Galleon layer for messaging integration with a remote ActiveMQ broker]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-13354[WFLY-13354]
+* https://issues.jboss.org/browse/WFLY-13798[WFLY-13798]
+
+=== Dev Contacts
+
+* mailto:ehugonne@redhat.com[Emmanuel Hugonnet]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly
+
+=== Other Interested Projects
+
+ActiveMQ Artemis
+
+== Requirements
+
+=== Hard Requirements
+
```

Comment:
@ehsavoie IIUC, provisioning a server with the `remote-activemq` layer, allows to connect to a remote ActiveMQ server just by configuring host, port, user, password (and then the user can get the JMS connection via JNDI `java:jboss/DefaultJMSConnectionFactory`) and there is **no need to add any resource adapter for the ActiveMQ Artemis server**: is that correct?

Reply:
yes that is correct